### DISCUSSION TITLE
could reducing memory-aliasing ,be a goal ,as to unlock further optimizations.

### DISCUSSION THREAD

**Discussion by gintominto5329** [carbon-language/carbon-lang#3604] [Language design]
hi,

avoiding memory-aliasing is one of the biggest reasons for standard FORTRAN ,to beat equivalent c-lang code
is carbon working to tackle this problem of c[++]?

thanks

**Comment by chandlerc** [comment]
Our goals already include performance. We try to have goals that are higher level than specific approaches though, and reducing memory-aliasing is just one specific approach.

FWIW, it's not clear to me to what extent the optimization capabilities of Fortran remain importantly superior to C++ these days -- a lot has changed about software, optimizing compilers, hardware, and C++ itself. But if Carbon makes it far enough for this kind of performance detail to be an area of focus (right now we're a bit focused on [C++ interop] instead), I suspect the community would be interested in proposals that try to address any important gaps here.

[C++ interop]: docs/project/roadmap.md#objective-for-2024-a-working-toolchain-that-supports-c-interop
